crates/admin/src/cluster_controller/logs_controller.rs

```diff
@@ -304,49 +305,196 @@ fn build_new_replicated_loglet_configuration(
-        let mut nodeset = NodeSet::empty();
+impl<'a> LogletPlacementSelector<'a> {
+    fn new<'b: 'a>(
+        nodes_config: &'b NodesConfiguration,
+        cluster_state: &'b ObservedClusterState,
+    ) -> LogletPlacementSelector<'a> {
+        Self {
+            nodes_config,
+            cluster_state,
+        }
+    }
 
-        for node in &log_servers {
-            nodeset.insert(node.as_plain());
+    /// Picks a set of storage nodes for a replicated loglet out of the available pool. Only writable
+    /// nodes will be considered.
+    fn select<R: Rng + ?Sized>(
+        &self,
+        strategy: PlacementStrategy,
+        replication_property: &ReplicationProperty,
+        rng: &mut R,
+        preferred_nodes: &NodeSet,
+    ) -> Result<NodeSet, PlacementError> {
+        if replication_property.at_greatest_scope().0 != &LocationScope::Node {
+            // todo: add support for zonal and regional placement
+            unimplemented!("only node-scoped replication placement is currently supported");
         }
 
-        Some(ReplicatedLogletParams {
-            loglet_id,
-            sequencer: **log_servers
-                .iter()
-                .choose(&mut rng)
-                .expect("one node must exist"),
-            replication,
-            nodeset,
-            write_set: None,
-        })
-    } else if let Some(sequencer) = log_servers.iter().choose(&mut rng) {
-        previous_configuration.cloned().map(|mut configuration| {
-            configuration.loglet_id = loglet_id;
-            configuration.sequencer = **sequencer;
-            configuration
-        })
-    } else {
-        None
+        // Only consider alive, writable nodes.
+        let candidates = WritableNodeSet::from(self.cluster_state, self.nodes_config);
+
+        if candidates.len() < replication_property.num_copies().into() {
+            return Err(PlacementError::InsufficientWriteableNodes);
+        }
+
+        let placement = match strategy {
+            PlacementStrategy::AllAvailable => NodeSet::from_iter(candidates),
+            PlacementStrategy::CappedNodesetSize(size) => {
+                let mut nodes = preferred_nodes
+                    .iter()
+                    .copied()
+                    .choose_multiple(rng, size.get() as usize);
+
+                if nodes.len() < size.get() as usize {
+                    let remaining = size.get() as usize - nodes.len();
+                    nodes.extend(candidates.into_iter().choose_multiple(rng, remaining));
+                }
+
+                NodeSet::from_iter(nodes)
+            }
+        };
+
+        let mut checker = NodeSetChecker::new(&placement, self.nodes_config, replication_property);
+        checker.set_attribute_on_each(&placement, || true);
+        if !checker.check_write_quorum(|attr| *attr) {
+            return Err(PlacementError::InsufficientWriteableNodes);
+        }
```

Comment:
Just checking for write quorum will not be enough to ensure that a minimum nodeset size is respected. Since we need read and write availability for sealing, I would suggest aiming for 2f + 1 node set size and f + 1 replication. With node set size being the replication factor, we will lose the capability to seal the log on the first node failure.